pkg/apis/core/validation/validation_test.go

```diff
@@ -18739,19 +18782,75 @@ func TestValidateTopologySpreadConstraints(t *testing.T) {
 	}
 
 	for i, tc := range testCases {
-		errs := validateTopologySpreadConstraints(tc.constraints, field.NewPath("field"))
+		t.Run(tc.name, func(t *testing.T) {
+			errs := validateTopologySpreadConstraints(tc.constraints, field.NewPath("field"))
+
+			if len(errs) > 0 && tc.errtype == "" {
+				t.Errorf("[%d: %q] unexpected error(s): %v", i, tc.name, errs)
+			} else if len(errs) == 0 && tc.errtype != "" {
+				t.Errorf("[%d: %q] expected error type %v", i, tc.name, tc.errtype)
+			} else if len(errs) >= 1 {
+				if errs[0].Type != tc.errtype {
+					t.Errorf("[%d: %q] expected error type %v, got %v", i, tc.name, tc.errtype, errs[0].Type)
+				} else if !strings.HasSuffix(errs[0].Field, "."+tc.errfield) {
+					t.Errorf("[%d: %q] expected error on field %q, got %q", i, tc.name, tc.errfield, errs[0].Field)
+				}
+			}
```

Comment:
It doesn't look like we are checking that the errors match in all cases.

Reply:
I change `TestValidateTopologySpreadConstraints` to simply compare the result with the expected value.